hardhat.config.zksync.ts

```diff
@@ -0,0 +1,157 @@
+import "module-alias/register";
+
+import "@matterlabs/hardhat-zksync";
+import "@matterlabs/hardhat-zksync-deploy";
+import "@matterlabs/hardhat-zksync-solc";
+import "@matterlabs/hardhat-zksync-verify";
+import "@nomicfoundation/hardhat-chai-matchers";
+import "@typechain/hardhat";
+import "hardhat-deploy";
+import { HardhatUserConfig, extendConfig } from "hardhat/config";
+import { HardhatConfig } from "hardhat/types";
+import "solidity-coverage";
+import "solidity-docgen";
+
+require("hardhat-contract-sizer");
+
+require("dotenv").config();
+
+const DEPLOYER_PRIVATE_KEY = process.env.DEPLOYER_PRIVATE_KEY;
+
+extendConfig((config: HardhatConfig) => {
+  if (process.env.EXPORT !== "true") {
+    config.external = {
+      ...config.external,
+      deployments: {
+        zksyncsepolia: [],
+      },
+    };
+  }
+});
+
+const config: HardhatUserConfig = {
+  defaultNetwork: "hardhat",
+  zksolc: {
+    version: "1.5.1",
+    compilerSource: "binary",
+    settings: {
+      metadata: {
+        // do not include the metadata hash, since this is machine dependent
+        // and we want all generated code to be deterministic
+        // https://docs.soliditylang.org/en/v0.7.6/metadata.html
+        bytecodeHash: "none",
+      },
+    },
+  },
+  solidity: {
+    compilers: [
+      {
+        version: "0.5.16",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
+          },
+          outputSelection: {
+            "*": {
+              "*": ["storageLayout"],
+            },
+          },
+        },
+      },
+      {
+        version: "0.8.25",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 10000,
+          },
+          evmVersion: "paris",
+          outputSelection: {
+            "*": {
+              "*": ["storageLayout"],
+            },
+          },
+        },
+      },
+    ],
+  },
+  zksolc: {
```

Comment:
zksolc object is already defined above, is there any logic behind defining it again.